wandb/sdk/wandb_init.py

```diff
@@ -136,16 +137,8 @@ def setup(self, kwargs) -> None:  # noqa: C901
         # Start with settings from wandb library singleton
         settings: Settings = self._wl.settings.copy()
         settings_param = kwargs.pop("settings", None)
-        if settings_param is not None:
-            if isinstance(settings_param, Settings):
-                # todo: check the logic here. this _only_ comes up in tests?
-                # update settings with settings_param using whatever
-                # source each parameter has there
-                settings._apply_settings(settings_param, _logger=logger)
```

Comment:
what about this function `_apply_settings` it is not relevant here anymore?
(I see it is also used in `wandb_setup` and `wandb_login`)

Reply:
Right, not used here anymore as the logic changed, but still used in other places. 
(`_apply_settings` applies settings to the target object using the Sources in the "source" object, so only the settings with greater-than-or-equal-to priorities will "pass"; in this case we want to explicitly apply `Source.INIT` as per user expectations)

Reply:
Got it! Thanks for the explanation